### DISCUSSION TITLE
[SOLVED] Interface without endpoints?

### DISCUSSION THREAD

**Discussion by KWolfe81** [hathach/tinyusb#948] [Q&A] [answered]
Any recommendations on how to add an interface without any endpoints?  Need it to support some legacy software applications.  I assume I'll need to be hacking up the device vendor class, but wanted to see if anyone had any pointers before I start.

**Comment by hathach** [comment]
For now you could implement your own driver using the `usbd_app_driver_get_cb()` callback, there is an example for doing so mentioned in the repo readme. For future, I will update the vendor driver to optionally accept 0 - 2 endpoints

**Comment by KWolfe81** [comment, marked as answer]
Thanks!  I'll go ahead and do that.  Modifying the existing Vendor Device code was straightforward enough - just literally commenting out the following in vendor_device.c. As always, thanks for a fantastic library.

```  TU_ASSERT(usbd_open_edpt_pair(rhport, tu_desc_next(itf_desc), 2, TUSB_XFER_BULK, &p_vendor->ep_out, &p_vendor->ep_in), 0);

  p_vendor->itf_num = itf_desc->bInterfaceNumber;

  // Prepare for incoming data
  if ( !usbd_edpt_xfer(rhport, p_vendor->ep_out, p_vendor->epout_buf, sizeof(p_vendor->epout_buf)) )
  {
    TU_LOG1_FAILED();
    TU_BREAKPOINT();
  }`
